stan/math/prim/fun/linspaced_int_array.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef STAN_MATH_PRIM_FUN_LINSPACED_INT_ARRAY_HPP
+#define STAN_MATH_PRIM_FUN_LINSPACED_INT_ARRAY_HPP
+
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return an array of linearly spaced integers.
+ *
+ * This produces an array from low to high (included) with integers spaced
+ * as (high - low) / (K - 1). For K=1, the array will contain the high value;
+ * for K=0 it returns an empty array.
+ *
+ * @param K size of the array
+ * @param low smallest value
+ * @param high largest value
+ * @return An array of size K with elements linearly spaced between
+ * low and high.
+ * @throw std::domain_error if K is negative, if high is less than low
+ * or if ().
+ */
+inline std::vector<double> linspaced_int_array(int K, int low, int high) {
+  static const char* function = "linspaced_int_array";
+  check_nonnegative(function, "size", K);
+  check_greater_or_equal(function, "high", high, low);
+  if (K == 0) {
+    return {};
+  }
+
+  Eigen::VectorXi v = Eigen::VectorXi::LinSpaced(K, low, high);
```

Comment:
Build the `std::vector<int>` directly.

Some variation of this I guess:
```c++
if(K == 1) {
  return { high };
}

std::vector<int> v(K);
int d = (high - low) / (K - 1);
for(size_t k = 0; k < K; ++k) {
  v[k] = k * d + low;
}
return v;
```

Reply:
Is there an advantage to that approach? This is the same impl. as the other linspaced function. But can also do that.

Reply:
It avoids two allocates (the local `std::vector<int>` will get returned directly by compiler optimizing returns) and the allocates and copies will be the expensive part of this.

But in the spirit of not tripping up on edge cases (that implementation I gave I think doesn't work for some of the examples I see [here](http://eigen.tuxfamily.org/dox-devel/classEigen_1_1DenseBase.html#title49)), I'll also take ```return {v.data(), v.data() + K};```

Reply:
Made it return {v.data(), v.data() + K};